### PR TITLE
[ADD] partner_route: New module

### PR DESCRIPTION
**Pull request by chienandalu**
This module serves as a baseline to define the partner routes that the members
of an organization need to do to acomplish a task over customers regularly

For example:

    - Commercial routes.
    - Delivery routes.
    - Salesmen routes.
    - Repair routes.

The module comes without any of this specific capacities, that should be further
developed.

image

cc @Tecnativa TT25235

### CODE REVIEW

**Comment by chienandalu** [comment]
@carlosdauden @sergio-teruel can you review?

**Review by carlosdauden — APPROVED** [review]
I miss some filter from route to see in which route a partner is or from partners to determine which partners are in a route, but nothing that is not solved with an advanced search.

File: partner_route/models/partner_route.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    user_id = fields.Many2one(
+        comodel_name="res.users",
+        string="Responsible",
+        help="Which user is in charge of the route",
+    )
+    route_type = fields.Selection(
+        selection=[
+            ("crm", "CRM"),
+            ("sale", "Sales"),
+            ("delivery", "Delivery"),
```
Comment (sergio-teruel) [inline partner_route/models/partner_route.py:23]: Maybe CRM and sale is the same...

File: partner_route/models/partner_route.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        related="partner_id.street",
+    )
+    city = fields.Char(
+        related="partner_id.city",
+    )
+    zip = fields.Char(
+        related="partner_id.zip",
+    )
+    state_id = fields.Many2one(
+        related="partner_id.state_id",
```
Comment (sergio-teruel) [inline partner_route/models/partner_route.py:96]: I am not sure... but comodel_name is not needed??

Comment (sergio-teruel) [reply partner_route/models/partner_route.py:96]: And the field strings??

**Review by sergio-teruel — CHANGES_REQUESTED** [review]

File: partner_route/models/partner_route.py (hunk 2 above)
Comment (pedrobaeza) [reply partner_route/models/partner_route.py:96]: On related fields, if not specified, they are taken from the source, so in summary, no need.

**Review by sergio-teruel — APPROVED** [review]

**Review by marcelsavegnago — APPROVED** [review]

**Pull request merged**